api/forwarder.go

```diff
@@ -128,21 +137,26 @@ func (f *Fwd) HandleProxyRequestByNodeID(targetAddress string, req *http.Request
 	return true, nil
 }
 
-func (f *Fwd) GetHTTPAddressByNodeID(getNodeID OwnerIDFunc) GetTargetAddressFunc {
-	return func(req *http.Request) (string, error) {
+func (f *Fwd) GetHTTPAddressByNodeID(getNodeID OwnerIDFunc) ParametersGetFunc {
+	return func(req *http.Request) (map[string]string, error) {
 		nodeID, err := getNodeID(req)
 		if err != nil {
-			return "", errors.Wrap(err, "fail to get target node ID")
+			return nil, errors.Wrap(err, "fail to get target node ID")
+		}
+		address, err := f.locator.Node2APIAddress(nodeID)
+		if err != nil {
+			return nil, errors.Wrap(err, "fail to get the address from node ID")
 		}
-		return f.locator.Node2APIAddress(nodeID)
+		return map[string]string{ParameterKeyAddress: address}, nil
 	}
 }
 
 const (
 	BackingImageUpload = "upload"
 )
 
-func (f *Fwd) HandleProxyRequestForBackingImageUpload(targetAddress string, req *http.Request) (proxyRequired bool, err error) {
+func (f *Fwd) HandleProxyRequestForBackingImageUpload(parameters map[string]string, req *http.Request) (proxyRequired bool, err error) {
+	targetAddress := parameters[ParameterKeyAddress]
 	if targetAddress == req.Host {
 		return false, fmt.Errorf("backing image upload request should not be handled by current longhorn manager service")
 	}
```

Comment:
Why do we need this check here?
Will the function return error as exepceted later if `targetAddress == req.Host` and without this check?

Reply:
This function is executed in the longhorn-manager pods and the `req.Host` is the manager pod IP. While the `targetAddress` should be the BIDS pod.
By explicitly erroring out here, I will know what happens...
If we didn't do anything, it will fallback to run the last function in this line. I may need to improve this by using a different function rather than `backingImageGet`: https://github.com/shuo-wu/longhorn-manager/blob/faa879b5ab35ce8f897c6b651c0c4592830de425/api/router.go#L147

Reply:
Understood. Thank you.